packages/hoprd/crates/hoprd-inbox/src/inbox.rs

```diff
@@ -16,6 +16,9 @@ pub trait InboxBackend<T: Copy + Default, M: Clone> {
     /// Create new storage with the given capacity and the timestamping function
     fn new_with_capacity(cap: usize, ts: TimestampFn) -> Self;
 
+    /// Either returns the tag, or get the tag of the oldest entry in the queue.
+    fn tag_resolution(&mut self, tag: Option<T>) -> Option<T>;
```

Comment:
Please do not put this fn into the trait, but rather make it a private method of the type that implements the `InboxBackend` trait.
Tag resolution is not something that should be on a public interface.